docs/docs/guide/strapi.mdx

```diff
@@ -0,0 +1,532 @@
+├── app/
+│   └── page.tsx               # Main page component handling experiment rendering
+├── lib/
+│   ├── types.ts               # TypeScript definitions for Strapi and experiments
+│   ├── strapi.ts              # Strapi API integration helper
+│   ├── growthbookServer.ts    # Server-side GrowthBook configuration
+│   ├── getVariationHelper.ts  # Helper for retrieving correct content variation
+│   └── GrowthbookTracking.tsx # Client-side experiment tracking
+├── middleware.ts              # UUID generation and cookie management
+└── .env
+```
+
+### Strapi Integration
+
+Let’s add some types we’ll need by adding a `lib` folder to the root of the project. Inside the new folder, create a `types.ts` file. (If you're not using TypeScript, skip this part.)
+
+```ts
+// /lib/types.ts
+type Variation = {
+  __component: string;
+  image?: {
+    url: string;
+    width: number;
+    height: number;
+    alternativeText: string;
+  };
+  [key: string]: unknown;
+};
+
+type ExperimentData = {
+  id: number;
+  featureFlagId?: string;
+  variations?: Variation[];
+  [key: string]: unknown;
+};
+
+type StrapiResponse = {
+  data: ExperimentData | ExperimentData[];
+};
+
+export type { Variation, ExperimentData, StrapiResponse };
+```
+
+These types make working with Strapi and GrowthBook content easier.
+
+### GrowthBook Integration
+
+Create a helper to fetch the correct variation from Strapi based on the experiment.
+
+In `lib`, create `getVariationHelper.ts`:
+
+```tsx
+// /lib/getVariationHelper.ts
+import type { GrowthBook } from "@growthbook/growthbook";
+import type { ExperimentData, StrapiResponse } from "./types";
+
+function isExperimentData(
+  data: ExperimentData
+): data is Required<ExperimentData> {
+  return "featureFlagId" in data && "variations" in data;
+}
+
+type Props = {
+  gb: GrowthBook;
+  growthbookExperiment: StrapiResponse | null;
+  featureFlagId: string;
+};
+
+/**
+ * Retrieves the variation for a given feature flag from a GrowthBook experiment.
+ *
+ * @param {Object} params - The parameters for the function.
+ * @param {any} params.gb - The GrowthBook instance used to evaluate the feature flag.
+ * @param {any} params.growthbookExperiment - The experiment data containing variations.
+ * @param {string} params.featureFlagId - The ID of the feature flag to retrieve the variation for.
+ */
+export default function getVariation({
+  gb,
+  growthbookExperiment,
+  featureFlagId,
+}: Props) {
+  if (!growthbookExperiment?.data) {
+    return null;
+  }
+
+  const experimentData = Array.isArray(growthbookExperiment.data)
+    ? growthbookExperiment.data.find((el) => el.featureFlagId === featureFlagId)
+    : growthbookExperiment.data;
+
+  if (!experimentData || !isExperimentData(experimentData)) {
+    return null;
+  }
+
+  // Evaluate the feature flag in the current context to get the variation index
+  const result = gb.evalFeature(featureFlagId).experimentResult;
+  const variationIndex = result?.variationId;
```

Comment:
I don't like this.  It's using a feature flag, but then only works if you add a specific experiment rule.  Also, the feature flag values are ignored.  I don't think we want to encourage this.

Some alternatives:

1. Skip feature flags entirely and use inline experiments instead with `gb.run(...)`.  https://docs.growthbook.io/lib/js#inline-experiments
2. Use a `number` feature flag and store the array index as the value.
3. Instead of an array of children in Stripe, store an array of objects, each with an id.  Use those ids as the feature flag values.

Reply:
Makes sense! What do you think about using option 3:

## Strapi
In Strapi, create a GrowthBook container that has a `featureFlagId` that connects to GrowthBook and a `variations` slot, in which you can put whatever components you want, as long as they have a `variationId`. 

![CleanShot 2024-12-21 at 17 45 23](https://github.com/user-attachments/assets/3be16392-ecbb-4dbb-9554-f45568a8012e)

## GrowthBook
Use a feature flag with `number` that maps to the `variationId` and `featureFlagId` above.

![CleanShot 2024-12-21 at 17 50 41](https://github.com/user-attachments/assets/018af69a-317d-4252-9e30-9f5998309d66)

## Next
The `getVariation` function returns `null` if the feature flag isn't live, otherwise it maps the component `variationId` from Strapi to the flag evaluation result from GrowthBook. This lets users employ feature flagging and experiments.
![CleanShot 2024-12-21 at 17 51 49](https://github.com/user-attachments/assets/10d91a66-26a5-45b5-a7c4-7dd6795c229d)